CMakeLists.txt

```diff
@@ -317,9 +317,9 @@ endif()
 # Threads (required)
 find_package(Threads REQUIRED)
 
-# Eigen (required)
-find_package(Eigen 3.3 REQUIRED)
-include_directories(SYSTEM ${EIGEN_INCLUDE_DIRS})
+# Eigen3 (required)
+find_package(Eigen3 3.3 REQUIRED NO_MODULE)
+include_directories(SYSTEM ${Eigen3_INCLUDE_DIR})
```

Comment:
Instead of adding the include as to everything, we should make use of the Eigen3::eigen targets and use target_link_libraries towards those that use it. ie. common.

Target names, might be different  I'm just guessing 😄

Reply:
See https://eigen.tuxfamily.org/dox/TopicCMakeGuide.html for example.

Reply:
Try add it to [here](https://github.com/PointCloudLibrary/pcl/pull/5613/files#diff-a77d393383a84e0dd42ebd261b80443390335377813ca160e32abf8c52effac2L178)